src/pip/_internal/resolution/resolvelib/candidates.py

```diff
@@ -107,8 +97,13 @@ def dist(self):
         return self._dist
 
     def get_dependencies(self):
-        # type: () -> Sequence[InstallRequirement]
-        return [self._make_install_req(str(r)) for r in self.dist.requires()]
+        # type: () -> Sequence[Requirement]
+        return [
+            self._factory.make_requirement(
+                self._factory.make_install_req(str(r), self._ireq),
```

Comment:
We can probably just merge `factory.make_requirement(factory.make_install_req(spec, ireq))` into `factory.make_requirement(spec, ireq)`.

Reply:
There is still need to build a `Requirement` from `InstallRequirement` (the initial requirements are passed in as `InstallRequirement`), but it’s definitely a good idea to have a helper function.

Reply:
I added a `make_requirement_from_spec()` to wrap the two steps. The original `make_requirement()` is renamed to `make_requirement_from_install_req()` to distinguish them. `make_install_req` is made private on factory since it is not used directly from outside anymore.